lib/action.ml

```diff
@@ -194,23 +243,32 @@ module Action (Github_api : Api.Github) (Slack_api : Api.Slack) = struct
     match ignore_notifications_from_user cfg req with
     | true -> Lwt.return []
     | false ->
-    match req with
-    | Github.Push n ->
-      partition_push cfg n |> List.map (fun (channel, n) -> generate_push_notification n channel) |> Lwt.return
-    | Pull_request n -> partition_pr cfg n |> List.map (generate_pull_request_notification n) |> Lwt.return
-    | PR_review n -> partition_pr_review cfg n |> List.map (generate_pr_review_notification n) |> Lwt.return
-    | PR_review_comment n ->
-      partition_pr_review_comment cfg n |> List.map (generate_pr_review_comment_notification n) |> Lwt.return
-    | Issue n -> partition_issue cfg n |> List.map (generate_issue_notification n) |> Lwt.return
-    | Issue_comment n -> partition_issue_comment cfg n |> List.map (generate_issue_comment_notification n) |> Lwt.return
-    | Commit_comment n ->
-      let%lwt channels, api_commit = partition_commit_comment ctx n in
-      let notifs = List.map (generate_commit_comment_notification api_commit n) channels in
-      Lwt.return notifs
-    | Status n ->
-      let%lwt channels = partition_status ctx n in
-      let notifs = List.map (generate_status_notification cfg n) channels in
-      Lwt.return notifs
+      let slack_match_func = match_github_login_to_slack_id (Some cfg) in
```

Comment:
Maybe we can move this outside of the match expression to not introduce the extra indentation?